src/wallet/rpcwallet.cpp

```diff
@@ -4662,7 +4662,9 @@ UniValue z_shieldcoinbase(const UniValue& params, bool fHelp)
                 }
             },
             [&](const libzcash::UnifiedAddress& ua) {
-                throw JSONRPCError(RPC_VERIFY_REJECTED, "Cannot shield coinbase output to a unified address.");
+                if (!Params().GetConsensus().NetworkUpgradeActive(nextBlockHeight, Consensus::UPGRADE_NU5)) {
+                    throw JSONRPCError(RPC_VERIFY_REJECTED, "Cannot shield coinbase output to a unified address.");
```

Comment:
```suggestion
                    throw JSONRPCError(RPC_VERIFY_REJECTED, "Cannot shield coinbase output to a unified address because NU5 is not active in the next block.");
```

Reply:
This is confusing though. Specifying a UA shouldn't require NU5; only shielding to an Orchard address should require NU5.

Reply:
Agreed. The behaviour I'd expect is that users can specify a UA _now_, and it will shield to the Sapling component of the UA (since the wallet should select the most preferred address that it is capable of sending to).